drivers/meter/ade9153a/ade9153a.h

```diff
@@ -0,0 +1,1471 @@
+
+/* ADE9153A_REG_PHSIGN Bit Definition */
+#define ADE9153A_CF2SIGN_MSK	            NO_OS_BIT(7)
+#define ADE9153A_CF1SIGN_MSK                NO_OS_BIT(6)
+#define ADE9153A_AVARSIGN_MSK	            NO_OS_BIT(1)
+#define ADE9153A_AWSIGN_MSK	                NO_OS_BIT(0)
+
+/* ADE9153A_REG_CONFIG2 Bit Definition */
+#define ADE9153A_UPERIOD_SEL_MSK	        NO_OS_BIT(12)
+#define ADE9153A_HPF_CRN_MSK                NO_OS_GENMASK(11, 9)
+
+/* ADE9153A_REG_EP_CFG Bit Definition */
+#define ADE9153A_NOLOAD_TMR_MSK	            NO_OS_GENMASK(7, 5)
+#define ADE9153A_RD_RST_EN_MSK	            NO_OS_BIT(3)
+#define ADE9153A_EGY_LD_ACCUM_MSK	        NO_OS_BIT(2)
+#define ADE9153A_EGY_TMR_MODE_MSK	        NO_OS_BIT(1)
+#define ADE9153A_EGY_PWR_EN_MSK	            NO_OS_BIT(0)
+
+/* ADE9153A_REG_CRC_FORCE Bit Definition */
+#define ADE9153A_FORCE_CRC_UPDATE_MSK	    NO_OS_BIT(0)
+
+/* ADE9153A_REG_TEMP_CFG Bit Definition */
+#define ADE9153A_TEMP_START_MSK	            NO_OS_BIT(3)
+#define ADE9153A_TEMP_EN_MSK	            NO_OS_BIT(2)
+#define ADE9153A_TEMP_TIME_MSK	            NO_OS_GENMASK(1, 0)
+
+/* ADE9153A_REG_TEMP_RSLT Bit Definition */
+#define ADE9153A_TEMP_RESULT_MSK	        NO_OS_GENMASK(11, 0)
+
+/* ADE9153A_REG_AI_PGAGAIN Bit Definition */
+#define ADE9153A_AI_SWAP_MSK	            NO_OS_BIT(4)
+#define ADE9153A_AI_GAIN_MSK	            NO_OS_GENMASK(2, 0)
+
+/* ADE9153A_REG_MS_STATUS_IRQ Bit Definition */
+#define ADE9153A_MS_SYSRDY_MSK	            NO_OS_BIT(14)
+#define ADE9153A_MS_CONFERR_MSK	            NO_OS_BIT(13)
+#define ADE9153A_MS_ABSENT_MSK	            NO_OS_BIT(12)
+#define ADE9153A_MS_TIMEOUT_MSK	            NO_OS_BIT(3)
+#define ADE9153A_MS_READY_MSK	            NO_OS_BIT(1)
+#define ADE9153A_MS_SHIFT_MSK	            NO_OS_BIT(0)
+
+/* ADE9153A_REG_EVENT_STATUS Bit Definition */
+#define ADE9153A_OIB_MSK	                NO_OS_BIT(5)
+#define ADE9153A_OIA_MSK	                NO_OS_BIT(4)
+#define ADE9153A_SWELLA_MSK	                NO_OS_BIT(2)
+#define ADE9153A_DIPA_MSK	                NO_OS_BIT(0)
+
+/* ADE9153A_REG_CHIP_STATUS Bit Definition */
+#define ADE9153A_UART_RESET_MSK	            NO_OS_BIT(7)
+#define ADE9153A_UART_ERROR2_MSK	        NO_OS_BIT(6)
+#define ADE9153A_UART_ERROR1_MSK	        NO_OS_BIT(5)
+#define ADE9153A_UART_ERROR0_MSK	        NO_OS_BIT(4)
+#define ADE9153A_ERROR3_MSK	                NO_OS_BIT(3)
+#define ADE9153A_ERROR2_MSK	                NO_OS_BIT(2)
+#define ADE9153A_ERROR1_MSK	                NO_OS_BIT(1)
+#define ADE9153A_ERROR0_MSK	                NO_OS_BIT(0)
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+/**
+ * @enum ade9153a_bi_gain_e
+ * @brief ADE9153A Bi Gain.
+ */
+enum ade9153a_bi_gain_e {
+	/* PGA gain for Current Channel B */
+	/* Gain 1 */
+	ADE9153a_PGA_CHB_GAIN_1 = 0,
```

Comment:
if there are only consecutive values in an enum, starting from 0, then there is no need for explicit initialization.